sql_server/pyodbc/base.py

```diff
@@ -201,7 +201,7 @@ def __init__(self, *args, **kwargs):
                     ops[op] = '%s COLLATE %s' % (sql, collation)
             self.operators.update(ops)
 
-        self.features = DatabaseFeatures(self)
+        self.features = DatabaseFeatures(self, opts.get('supports_nullable_unique_constraints', False))
```

Comment:
Wouldn't it make more sense to just check `connection.settings_dict['options']` within `DatabaseFeatures.__init__`? That way you don't need to change the signature around for every custom feature that's implemented.